libp2p/autotls/utils.nim

```diff
@@ -53,19 +53,10 @@ proc isPublic*(ip: IpAddress): bool {.raises: [AutoTLSError].} =
     raise newException(AutoTLSError, "Failed to parse IP address", exc)
 
 proc getPublicIPAddress*(): IpAddress {.raises: [AutoTLSError].} =
-  try:
-    let ip = checkedGetPrimaryIPAddr()
-    if not ip.isIPv4():
-      raise newException(AutoTLSError, "Host does not have an IPv4 address")
-    if not ip.isPublic():
-      raise newException(AutoTLSError, "Host does not have a public IPv4 address")
-    return ip
-  except AutoTLSError as exc:
-    raise exc
-  except CatchableError as exc:
-    raise newException(
-      AutoTLSError, "Unexpected error while getting primary IP address for host", exc
-    )
+  let ip = checkedGetPrimaryIPAddr()
+  assert ip.isIPv4(), "Host does not have an IPv4 address"
+  assert ip.isPublic(), "Host does not have a public IPv4 address"
```

Comment:
I'm not sure of these assertions, as these could be runtime errors.
It could happen for a device to move to diff network and end up then in one in which the assigned IP is not public. Then in that case we'd be crashing the whole app.